pkg/client/client_v2_test.go

```diff
@@ -869,6 +875,149 @@ func TestClientV2_Checkin_FeatureFlags(t *testing.T) {
 	require.Empty(t, errs)
 }
 
+type unitChangesAccumulator struct {
+	mx          sync.Mutex
+	unitChanges []UnitChanged
+}
+
+func (uca *unitChangesAccumulator) Start(ctx context.Context, t *testing.T, c <-chan UnitChanged) {
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case uc := <-c:
+				func() {
+					t.Logf("received unit change %v", uc)
+					switch uc.Type {
+					case UnitChangedAdded:
+						err := uc.Unit.UpdateState(UnitStateHealthy, "Healthy", map[string]interface{}{
+							"custom": "payload",
+						})
+						require.NoError(t, err)
+					}
+
+					uca.mx.Lock()
+					defer uca.mx.Unlock()
+
+					uca.unitChanges = append(uca.unitChanges, uc)
+				}()
+			}
+		}
+	}()
+}
+
+func (uca *unitChangesAccumulator) GetAccumulatedChanges() []UnitChanged {
+	uca.mx.Lock()
+	defer uca.mx.Unlock()
```

Comment:
Could these be `uca.mx.RLock()` and `uca.mx.RUnlock()` with `uc.mx` becoming an `RWMutex`?

Reply:
In the test there is exactly 1 writer (the accumulator) and 1 reader (the test itself): an RWMutex would make sense in case we have multiple readers and (maybe) multiple writers as we allow concurrent read access reducing contention...

For this case I think that a simple mutex is enough (we are not looking for performance or anything, just avoiding data races)